cli/flamingock-cli/src/main/java/io/flamingock/cli/config/DatabaseConfig.java

```diff
@@ -269,16 +270,34 @@ public void setPassword(String password) {
             this.password = password;
         }
 
-        public SqlDialect getSqlDialect() {
-            return sqlDialect;
+        public Optional<SqlDialect> getSqlDialect() {
+            return sqlDialect == null ? Optional.empty() :  Optional.of(sqlDialect);
         }
 
         public void setSqlDialect(String sqlDialect) {
             this.sqlDialect = SqlDialect.valueOf(sqlDialect.toUpperCase());
         }
 
+        public SqlDialect getEffectiveSqlDialect() {
+            if (this.sqlDialect != null) {
+                return this.sqlDialect;
+            }
+            String[] parts = this.endpoint.split(":", 3);
+            if (parts.length < 2 || parts[1].isEmpty()) {
+                throw new IllegalStateException("Cannot determine SQL dialect from endpoint: " + this.endpoint);
+            }
+            String dialect = parts[1].toLowerCase();
+            if ("firebirdsql".equals(dialect)) dialect = "firebird";
+            if ("informix-sqli".equals(dialect)) dialect = "informix";
+            try {
+                return SqlDialect.valueOf(dialect.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Unsupported SQL Dialect: " + dialect, e);
+            }
```

Comment:
This is probably better to be moved to the enum itself, to a method called something like : `SqlDialect.fromString()`.

The reason behind is that we can encapsulate the small logic around the dialect translation in one place.